enforcer-rules/src/main/java/com/google/cloud/tools/opensource/enforcer/LinkageCheckerRule.java

```diff
@@ -110,24 +125,41 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
       try {
         LinkageChecker linkageChecker = LinkageChecker.create(classpath, directDependencies);
         LinkageCheckReport linkageReport = linkageChecker.findLinkageErrors();
-        int totalErrors =
+        long errorCount =
             linkageReport.getJarLinkageReports().stream()
                 .mapToInt(JarLinkageReport::getCauseToSourceClassesSize)
                 .sum();
-        if (totalErrors > 0) {
+        if (reportOnlyReachable) {
+          ImmutableList<JarLinkageReport> reachableErrorReports =
+              linkageReport.getJarLinkageReports().stream()
+                  .map(JarLinkageReport::reachableErrors)
+                  .collect(toImmutableList());
+          linkageReport = LinkageCheckReport.create(reachableErrorReports);
+          long reachableErrorCount =
+              reachableErrorReports.stream()
+                  .mapToInt(JarLinkageReport::getCauseToSourceClassesSize)
+                  .sum();
+          errorCount = reachableErrorCount;
+        }
+
+        String foundError =
```

Comment:
This is hard to decode. Perhaps break it up into separate variables for the different cases inside the relevant if-else blocks.

Reply:
Updated this to simpler form.